drivers: ps2: Microchip XEC PS2 add MEC172x support

jvasanth1 opened the pull request:
Update the Microchip XEC PS2 driver to support MEC172x. Also add pinctrl support.

jvasanth1 committed:
drivers: ps2: Microchip XEC PS2 add MEC172x support

Update the Microchip XEC PS2 driver to support MEC172x.
NOTE: MEC15xx has two PS2 controllers and
MEC172x has one.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 committed:
drivers: ps2: Microchip XEC PS2 driver add PINCTRL support

Add optional PINCTRL support to the Microchip XEC PS2 driver
shared between MEC15xx and MEC172x families.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

MaureenHelm commented on drivers/ps2/Kconfig.xec:
```diff
@@ -19,6 +20,7 @@ config PS2_XEC_0
 
 config PS2_XEC_1
 	bool "PS2_XEC_1"
+	depends on SOC_SERIES_MEC1501X
```
`PS2_XEC_0` and `PS2_XEC_1` both need to be removed

jvasanth1 replied:
MEC15xx has 2 PS2 controllers and MEC17xx has only one. Hence the` depends on SOC_SERIES_MEC1501X` is not applicable `PS2_XEC_0`

MaureenHelm replied:
The number of instances is defined in the SoC-level devicetrees, and after the refactoring in this PR to use `DT_INST_FOREACH_STATUS_OKAY`, the driver isn't using these Kconfig symbols anymore.

jvasanth1 replied:
removed `PS2_XEC_0 `and `PS2_XEC_1 `symbols

MaureenHelm commented on boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig:
```diff
@@ -18,5 +18,6 @@ CONFIG_UART_CONSOLE=y
 CONFIG_ADC=y
 CONFIG_I2C=y
 CONFIG_ESPI=y
+CONFIG_PS2=y
```
This whole file needs to be cleaned up to include only the drivers needed for hello_world

jvasanth1 replied:
okay, will remove adc, i2c, espi, ps2, spi, spi_async

jvasanth1 replied:
including only drivers for hello_world

MaureenHelm reviewed (changes requested):
Please split the cleanup changes and MEC172x changes into separate commits. There are also some twister failures that need to be addressed.

jvasanth1 committed:
drivers: ps2: Standarize device tree and structure

Standardize PS2 device tree properties. Standardize device
structure usage. Abstract GIRQ and PCR register access.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 committed:
boards: arm: mec172xevb_assy6906: Include limited drivers

Include only required drivers for hello world application

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

jvasanth1 commented:
> Please split the cleanup changes and MEC172x changes into separate commits. There are also some twister failures that need to be addressed.

split into separate commits

jvasanth1 commented:
Any inputs on the twister error?

It seems to be a BOUNDS_CHECK_BYPASS_MITIGATION error. Not sure how to debug this.

```
warning: BOUNDS_CHECK_BYPASS_MITIGATION (defined at kernel/Kconfig:762) was assigned the value 'y'
but got the value 'n'.
```

jvasanth1 commented:
Hi @MaureenHelm I fixed the build errors. Please help to approve.

jvasanth1 commented on boards/arm/mec1501modular_assy6885/pinmux.c:
```diff
@@ -280,7 +280,7 @@ static int board_pinmux_init(const struct device *dev)
 #endif
 
-#ifdef CONFIG_PS2_XEC_0
+#if DT_NODE_HAS_STATUS(DT_INST(0, microchip_xec_ps2), okay)
```
fixed accordingly
